apps/verticalhome/js/pin_app_navigation.js

```diff
@@ -7,38 +7,57 @@
    * This is PinAppNavigation's constructor.
    * @param {Object} list [DOM container that contains pin elements]
    */
-  function PinAppNavigation(list) {
-    if (!list && typeof list !== 'object') {
-      console.error('List of pin apps must be an object');
-      return;
+  function PinAppNavigation(list, selector) {
+    if (!list) {
+      throw new Error('List must be a dom container ' +
+        'with elements to be navigated');
     }
 
-    this.elemList = list;
-    window.removeEventListener('keydown', this);
-    window.addEventListener('keydown', this);
-  }
+    /**
+     * This variable contains all elements that will
+     * be obtained through this.selector
```

Comment:
emm, can we have a bit better description here? :) something like:
"all elements that are representing applications on the screen"? since 'this.selector ' is not that obvious explanation of what kind of elements are gonna be stored here.